llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -1265,10 +1278,15 @@ bool SPIRVToLLVM::postProcessRowMajorMatrix() {
 
             // If we are loading a full row major matrix, need to load the rows and then transpose.
             for (unsigned i = 0; i < rowCount; i++) {
-              Value *pointerElem = getBuilder()->CreateGEP(
-                  pointerElemType, pointer,
-                  {getBuilder()->getInt32(0), getBuilder()->getInt32(i), getBuilder()->getInt32(0)});
-              Type *castType = pointerElem->getType()->getPointerElementType();
+              SmallVector<Value *, 3> indices;
+              indices.push_back(zero);
+              indices.push_back(getBuilder()->getInt32(i));
+              indices.push_back(zero);
```

Comment:
We don't seem to need vectors here at all? Can we just use initialiser list or `ArrayRef<>`?

Reply:
I'm running into some issues when using ArrayRef<> and "release" version of the code. It looks like "getBuilder()->getInt32" is not initializing object properly when invoked inside initialize_list.
Do you know about any limitations for ArrayRef<>

Reply:
ArrayRef and Twine cannot be stored, so you cannot have a local variable of them (the backing memory immediately goes out of scope). It only works if you inline the list into the function call argument, like it was before.

Reply:
> It looks like "getBuilder()->getInt32" is not initializing object properly when invoked inside initialize_list.

That's odd. If I read the code right, all the elements point to values stored somewhere else, so it should be safe to pass an `ArrayRef<>` or initialiser list.

Reply:
I returned to the initialization list for `CreateGEP` and added the same for `getIndexedType`. I also added comment because my intention was to have a single vector of indices and pass it to `CreateGEP` and `getIndexedType` which in my opinion is less error prone for later changes.

Reply:
Just in case: you can still do it your way by declaring `indices` an ordinary array, so `Value *indices[] = ...`, and passing it as an `ArrayRef<>`. Still no need for vectors.